pages/[[...slug]].js

```diff
@@ -1,27 +1,37 @@
 import DotCMSPage from '../components/layout/DotCMSPage';
 import Error from '../components/layout/Error';
-import { getPage, getNav, getPathsArray } from '../utilities/dotcms';
+import {
+    getPage,
+    getNav,
+    getPathsArray,
+    getLanguages,
+    getLanguagesProps
+} from '../utilities/dotcms';
 import getPageList from '../utilities/dotcms/getPageList';
 
-export default function Page({ pageRender, nav, error }) {
+export default function Page({ pageRender, nav, error, languageProps }) {
     if (error) {
         return <Error statusCode={error.statusCode} message={error.message} />;
     }
-    return <DotCMSPage pageRender={pageRender} nav={nav} />;
+    return <DotCMSPage pageRender={pageRender} nav={nav} languageProps={languageProps} />;
 }
 
 export const getStaticPaths = async () => {
     // Fetch pages from DotCMS and get all the urls in an array of strings, ex:
-    // ['/destinations/index', '/store/index', '/store/category/snow']
+    // ['/destinations/index', '/store/index', '/store/category/snow'];
+
     const pageList = await getPageList();
 
+    // Fetch list of languages supported in the DotCMS instance so we can build our static pages.
+    const { languages  } = await getLanguagesProps();
```

Comment:
We talked about moving this inside `getPageList` and return `['destination', '/en/destination']`, wasn't that possible? I think this approach is better and cleaner because in one place we get ALL the pages from dotcms (all pages in all languages).

Reply:
Not sure I'm following here. `getPageList` is a GraphQL query and it doesn't returns `/en/` paths for any route.